tests/ha/await_upgrade.pm

```diff
@@ -0,0 +1,32 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Wait for all nodes which are allowed to upgrade before.
+# Maintainer: Christian Lanig <clanig@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use hacluster;
+use lockapi;
+
+sub run {
+
+    my $cluster_name = get_cluster_name;
+    record_info("Upgrade node 1", "upgrade has started for node 1") if is_node(1);
+
+    if (is_node(2)) {
+        record_info("Waiting node 1", "Node 1 is upgrading");
+        barrier_wait("NODE_UPGRADED_${cluster_name}_NODE1");
+        record_info("Upgrade node 2", "Node 2 is upgrading");
```

Comment:
Maybe better to have the same message for all nodes? Not so important but it could be useful in case of text search in the job logs, no?

Reply:
> Maybe better to have the same message for all nodes? Not so important but it could be useful in case of text search in the job logs, no?

Actually, I think these messages should be different, and the one in L28 actually closer to the one in L22. L26 reports that node 2 is waiting as the next command is a `barrier_wait`, while L28 should report that node 2 is starting its upgrade *while* node 1 waits in `tests/ha/wait_others_upgraded`.